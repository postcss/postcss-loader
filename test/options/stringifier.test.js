import {
  compile,
  getCompiler,
  getErrors,
  getCodeFromBundle,
  getWarnings,
} from '../helpers/index';

describe('Options Stringifier', () => {
  it('should work Stringifier - {String}', async () => {
    const compiler = getCompiler('./css/index.js', {
      postcssOptions: {
        stringifier: 'sugarss',
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.css', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work Stringifier - {Object}', async () => {
    const compiler = getCompiler('./css/index.js', {
      postcssOptions: {
        // eslint-disable-next-line global-require
        stringifier: require('sugarss'),
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.css', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work Stringifier - {Function}', async () => {
    // eslint-disable-next-line global-require
    const Midas = require('midas');
    const midas = new Midas();

    const compiler = getCompiler('./css/index.js', {
      postcssOptions: {
        stringifier: midas.stringifier,
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.css', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should throw an error on "unresolved" stringifier', async () => {
    const compiler = getCompiler('./css/index.js', {
      postcssOptions: {
        stringifier: 'unresolved',
      },
    });
    const stats = await compile(compiler);

    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats, true)).toMatchSnapshot('errors');
  });
});
