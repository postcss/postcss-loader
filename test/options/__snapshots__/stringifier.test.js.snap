// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Stringifier should emit error Stringifier: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
TypeError: this.stringify is not a function",
  "ModuleError: Module Error (from \`replaced original path\`):
(Emitted value instead of an instance of Error) Loading PostCSS Stringifier failed: Cannot find module 'unresolved' from 'src/index.js'",
]
`;

exports[`Options Stringifier should emit error Stringifier: warnings 1`] = `Array []`;

exports[`Options Stringifier should work Stringifier - {Function}: css 1`] = `
"<pre class=\\"midas\\"><code><span class=\\"midas__selector\\"><span class=\\"midas__tag\\">a</span></span> <span class=\\"midas__brace\\">{</span> <span class=\\"midas__property\\">color</span><span class=\\"midas__colon\\">: </span><span class=\\"midas__value\\"><span class=\\"midas__word\\">black</span></span> <span class=\\"midas__brace\\">}</span>
</code></pre>"
`;

exports[`Options Stringifier should work Stringifier - {Function}: errors 1`] = `Array []`;

exports[`Options Stringifier should work Stringifier - {Function}: warnings 1`] = `Array []`;

exports[`Options Stringifier should work Stringifier - {Object}: css 1`] = `
"a  color: black
"
`;

exports[`Options Stringifier should work Stringifier - {Object}: errors 1`] = `Array []`;

exports[`Options Stringifier should work Stringifier - {Object}: warnings 1`] = `Array []`;

exports[`Options Stringifier should work Stringifier - {String}: css 1`] = `
"a  color: black
"
`;

exports[`Options Stringifier should work Stringifier - {String}: errors 1`] = `Array []`;

exports[`Options Stringifier should work Stringifier - {String}: warnings 1`] = `Array []`;
